sc-memory/sc-memory/sc_template_build.cpp

```diff
@@ -144,9 +144,13 @@ class ScTemplateBuilder
   friend class ScTemplate;
 
 protected:
-  ScTemplateBuilder(ScAddr const & inScTemplateAddr, ScMemoryContext & inCtx)
+  ScTemplateBuilder(
```

Comment:
Search algorithm should use parameters replacement. We should not change template during build. We should have one template, but putting different parameters in search we will get different results

Reply:
I agree and I tried to do it at first but I did not succeed. I will clarify what the difficulties were and we could discuss it